repos/system_upgrade/el7toel8/actors/selinux/selinuxcontentscanner/actor.py

```diff
@@ -0,0 +1,223 @@
+import os
+import re
+from shutil import rmtree
+
+from leapp.actors import Actor
+from leapp.models import SELinuxModules, SELinuxModule, SELinuxCustom, SELinuxFacts, SELinuxRequestRPMs, RpmTransactionTasks
+from leapp.tags import FactsPhaseTag, IPUWorkflowTag
+from leapp.libraries.stdlib import run, CalledProcessError
+
+WORKING_DIRECTORY = '/tmp/selinux/'
+
+# types and attributes that where removed between RHEL 7 and 8
+REMOVED_TYPES_=["base_typeattr_15","direct_run_init","gpgdomain","httpd_exec_scripts","httpd_user_script_exec_type","ibendport_type","ibpkey_type","pcmcia_typeattr_2","pcmcia_typeattr_3","pcmcia_typeattr_4","pcmcia_typeattr_5","pcmcia_typeattr_6","pcmcia_typeattr_7","sandbox_caps_domain","sandbox_typeattr_2","sandbox_typeattr_3","sandbox_typeattr_4","server_ptynode","systemctl_domain","user_home_content_type","userhelper_type","cgdcbxd_exec_t","cgdcbxd_t","cgdcbxd_unit_file_t","cgdcbxd_var_run_t","ganesha_use_fusefs","ganesha_exec_t","ganesha_t","ganesha_tmp_t","ganesha_unit_file_t","ganesha_var_log_t","ganesha_var_run_t","ganesha_use_fusefs"]
+# to be used with grep
+REMOVED_TYPES="|".join(REMOVED_TYPES_)
+# to be used with sed
+SED_COMMAND='/' + '\|'.join(REMOVED_TYPES_) + '/s/^/;/g'
+
+# types, attributes and boolean contained in container-selinux
+CONTAINER_TYPES="|".join(["container_connect_any","container_runtime_t","container_runtime_exec_t","spc_t","container_auth_t","container_auth_exec_t","spc_var_run_t","container_var_lib_t","container_home_t","container_config_t","container_lock_t","container_log_t","container_runtime_tmp_t","container_runtime_tmpfs_t","container_var_run_t","container_plugin_var_run_t","container_unit_file_t","container_devpts_t","container_share_t","container_port_t","container_build_t","container_logreader_t","docker_log_t","docker_tmpfs_t","docker_share_t","docker_t","docker_lock_t","docker_home_t","docker_exec_t","docker_unit_file_t","docker_devpts_t","docker_config_t","docker_tmp_t","docker_auth_exec_t","docker_plugin_var_run_t","docker_port_t","docker_auth_t","docker_var_run_t","docker_var_lib_t","container_domain","container_net_domain"])
+
+
+class SELinuxContentScanner(Actor):
+    '''
+    Scan the system for any SELinux customizations
+
+    Find SELinux policy customizations (custom policy modules and changes
+    introduced by semanage) and save them in SELinuxModules and SELinuxCustom
+    models. Customizations that are incompatible with SELinux policy on RHEL-8
+    are removed.
+    '''
+    name = 'selinuxcontentscanner'
+    consumes = (SELinuxFacts, )
+    produces = (SELinuxModules, SELinuxCustom, SELinuxRequestRPMs, RpmTransactionTasks,)
+    tags = (FactsPhaseTag, IPUWorkflowTag, )
+
+    def process(self):
+        # exit if SELinux is disabled
+        for fact in self.consume(SELinuxFacts):
+            if not fact.enabled:
+                return
+
+        (semodule_list, request_rpms) = self.getSELinuxModules()
```

Comment:
Maybe this will be more readable?
(semodule_list, rpms_to_keep, rpms_to_install) = self.getSELinuxModules()